point_cloud_client/src/lib.rs

```diff
@@ -98,10 +98,11 @@ impl<'a> PointCloudClientBuilder<'a> {
             .iter()
             .map(|location| self.data_provider_factory.generate_data_provider(location))
             .collect::<Result<Vec<Box<dyn DataProvider>>>>()?;
-        let mut aabb: Option<Aabb3<f64>> = None;
-        let unite = |bbox: &Aabb3<f64>, with: &mut Option<Aabb3<f64>>| {
-            let b = with.get_or_insert(*bbox);
-            *b = b.union(bbox);
+        let mut aabb: Option<Aabb<f64>> = None;
+        let unite = |bbox: &Aabb<f64>, with: &mut Option<Aabb<f64>>| {
+            let b = with.get_or_insert(bbox.clone());
+            b.grow(*bbox.min());
```

Comment:
I don't think it makes this code any easier to understand, since this function mutates, but I'd expect a `union` method to have an immutable signature, i.e. `fn(self, other: &Self) -> Self`.

I'm fine with implementing it if you prefer though.

Reply:
It's fine with me, you can also leave it..